app/models/authorization_request/france_connect.rb

```diff
@@ -18,6 +18,13 @@ class AuthorizationRequest::FranceConnect < AuthorizationRequest
 
   contact :contact_technique, validation_condition: ->(record) { record.need_complete_validation?(:contacts) }
 
+  validates :contact_technique_phone_number,
+    french_phone_number: {
+      france_connect_mobile: true,
+      message: :france_connect_mobile_required
+    },
+    if: ->(record) { record.need_complete_validation?(:contacts) }
```

Comment:
L'énorme avantage de ActiveModel::Errors est que tu n'as pas besoin de faire une clé spécifique pour FranceConnect, tu peux juste override dans les attributs pour FC l'erreur.

Donc ici:


```suggestion
  validates :contact_technique_phone_number,
    french_phone_number: {
      mobile: true,
    },
    if: ->(record) { record.need_complete_validation?(:contacts) }
```

Dans les locales (activerecord.yml)

```yaml
    errors:
      messages: *active_record_messages_errors
      models:
        authorization_request/france_connect: 
          attributes:
            contact_technique_phone_number:
              mobile: 'n'est pas un numéro de téléphone valide : afin de recevoir le SMS qui lui sera envoyé, le numéro du responsable technique doit être un numéro de portable.'
            
```

Et du coup tu peux simplifier le validateur (parce qu'en réalité c'est étrange de faire une validation spécifique à FC alors que c'est la même chose que l'option mobile), et tu peux garder ton test E2E (si tu veux) pour garantir l'affichage du bon texte.